### PR TITLE
Fix commit_sig encoding

### PR DESCRIPTION
**Pull request by oacdutra**
Hi @dr-orlovsky,

I tried to solve this issue related to [this comment](https://github.com/LNP-WG/lnp-core/issues/32#issuecomment-1493413664).

### CODE REVIEW

File: lnp2p/src/bolt/bolt2.rs [hunk 1]
```diff
@@ -735,9 +734,46 @@ impl DumbDefault for AcceptChannel {
...
+        let num_htlcs = self.htlc_signatures.len() as u16;
+        num_htlcs.lightning_encode(&mut e)?;
+        for index in 0..num_htlcs {
+            let sig: Signature = self.htlc_signatures[index as usize];
+            sig.lightning_encode(&mut e)?;
+        }
+
+        let buf: Vec<u8> = vec![];
+        e.write_all(&buf)?;
+        Ok(buf.len())
```
Comment (dr-orlovsky) [inline lnp2p/src/bolt/bolt2.rs]: Can you pls explain what this code tries to achieve?

File: lnp2p/src/bolt/bolt2.rs [hunk 2]
```diff
@@ -735,9 +734,46 @@ impl DumbDefault for AcceptChannel {
...
+    }
+}
+
+impl lightning_encoding::LightningDecode for CommitmentSigned {
+    fn lightning_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+        let channel_id = ChannelId::lightning_decode(&mut d)?;
+        let signature = Signature::lightning_decode(&mut d)?;
+        let num_htlcs = u16::lightning_decode(&mut d)?;
+
+        let mut htlc_signatures = vec![];
```
Comment (dr-orlovsky) [inline lnp2p/src/bolt/bolt2.rs]:
```suggestion
        let mut htlc_signatures = Vec::with_capacity(num_htlcs as usize);
```

File: lnp2p/src/bolt/bolt2.rs [hunk 3]
```diff
@@ -735,9 +734,46 @@ impl DumbDefault for AcceptChannel {
...
+        for index in 0..num_htlcs {
+            let sig = Signature::lightning_decode(&mut d)?;
+            htlc_signatures.insert(index.into(), sig);
+        }
```
Comment (dr-orlovsky) [inline lnp2p/src/bolt/bolt2.rs]:
```suggestion
        for _ in 0..num_htlcs {
            let sig = Signature::lightning_decode(&mut d)?;
            htlc_signatures.push(sig);
        }
```

File: lnp2p/src/bolt/bolt2.rs [hunk 4]
```diff
@@ -735,9 +734,46 @@ impl DumbDefault for AcceptChannel {
     }
 }
 
+impl lightning_encoding::LightningEncode for CommitmentSigned {
```
Comment (dr-orlovsky) [inline lnp2p/src/bolt/bolt2.rs:738]:
```suggestion
// TODO: Replace this custom implementation with derivation in v0.10
impl lightning_encoding::LightningEncode for CommitmentSigned {
```

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
I think this can work as a workaround. Some parts of code are strange (see comments) and `lightning_encode` also returns invalid number of bytes written. Otherwise LGTM

File: lnp2p/src/bolt/bolt2.rs (hunk 1 above)
Comment (oacdutra) [reply lnp2p/src/bolt/bolt2.rs]: Return the number of bytes.

Comment (dr-orlovsky) [reply lnp2p/src/bolt/bolt2.rs]: It will always return zero. It does nothing

**Review by dr-orlovsky — APPROVED** [review]
ACK https://github.com/LNP-WG/lnp-core/commit/5f09112ddb4734d1c552005e279674e09ecd5192

**Pull request merged**